src/Arcus.Messaging.EventHubs.Core/Extensions/AzureClientFactoryBuilderExtensions.cs

```diff
@@ -169,8 +176,21 @@ private static string GetEventHubsConnectionString(string connectionStringSecret
                     + "please use the 'services.AddSecretStore(...)' or 'host.ConfigureSecretStore(...)' (https://security.arcus-azure.net/features/secret-store)");
             }
 
-            string connectionString = secretProvider.GetRawSecretAsync(connectionStringSecretName).GetAwaiter().GetResult();
-            return connectionString;
+            try
+            {
+                string connectionString = secretProvider.GetRawSecret(connectionStringSecretName);
+                return connectionString;
+            }
+            catch (Exception exception)
+            {
+                ILogger logger = 
+                    serviceProvider.GetService<ILogger<EventHubProducerClient>>() 
+                    ?? NullLogger<EventHubProducerClient>.Instance;
+
+                logger.LogWarning(exception, "Cannot synchronously retrieve Azure EventHubs connection string secret for '{SecretName}', fallback on asynchronously", connectionStringSecretName);
```

Comment:
I would not log this as a warning, rather as a trace or debug.
If you do not have a secret provider registered that supports sync retrieval, then this is something that you're not going to 'fix'.  It's also not a real issue imho.

Reply:
Moved to `LogTrace` in https://github.com/arcus-azure/arcus.messaging/pull/373/commits/efbcb36c397809e914f3951916188d0c51fc8344